### PR TITLE
Add pipeline status endpoint

### PR DESCRIPTION
**Pull request by Mehrnaz-Charkhchi**
### Description
When the client asks, query the Seqera platform, and report the status of the workflow to the client.

### Related JIRA Issue(s)
[ENSWBSITES-2649](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2649)

### CODE REVIEW

File: app/core/config.py
Comment (kamaldodiya) [inline app/core/config.py:62]: There are two URLs for which base (HOST) will be same `SEQERA_WORKFLOW_LAUNCH_URL` and  `SEQERA_API`.
Would it be possible to just have `SEQERA_API` and remove `SEQERA_WORKFLOW_LAUNCH_URL` and define the LAUNCH_URL from `SEQERA_API`.
If not in this PR then may be in future

Comment (Mehrnaz-Charkhchi) [reply app/core/config.py:62]: I did think about it @kamaldodiya and I agree with you. I thought maybe after we get all the approvals for this PR I can make that update to reduce the noise.

File: app/vep/utils/nextflow.py
Comment (kamaldodiya) [inline app/vep/utils/nextflow.py:50]: If the purpose here is to log the connection issue while requesting seqera API then it would be good to be explicit and say connection issue with seqera API while logging

Comment (Mehrnaz-Charkhchi) [reply app/vep/utils/nextflow.py:50]: It's a bit more general than that because we're not connecting to seqera here. It can be any unexpected issue that happened while connecting to this API exactly like line 31.

**Review by kamaldodiya — APPROVED** [review]

File: app/vep/vep_resources.py
Comment (azangru) [inline app/vep/vep_resources.py]: Didn't we see that Seqera's response to a missing workflow id can also be a `400`? We would also want to transform that into a 404.

File: app/vep/vep_resources.py
Comment (azangru) [inline app/vep/vep_resources.py:81]: This isn't what I wrote in the spec 🤷‍♂️ Are we changing the spec?

File: app/vep/vep_resources.py
Comment (azangru) [inline app/vep/vep_resources.py:73]: I may be misreading the python here; but what's the shape of the response going to be? I don't think it conforms to the spec.

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:81]: In the spec for this specific error  we're drifting from the general error format that we use, I'm ready to either updating the spec or code but I thought you should know that before.

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py]: Done!

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:73]: What do you mean?
This response is not in the spec?
`{
    "job_id": "5SYhtjFhDfmzx2",
    "status": "FAILED"
}`

Comment (azangru) [reply app/vep/vep_resources.py:73]: The spec calls the field `submission_id` 😉

File: app/vep/vep_resources.py
Comment (azangru) [reply app/vep/vep_resources.py:81]: I did not know about the general error format. Personally, I think the status_code field in the body is excessive (the code is already present in the header); but if that's our general error format... Let's see what others think.

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:73]: Ah, I got confused by having two names for submission_id, in the url we call it job_id and in the response we call it submission_id! we should update one of them!

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:81]: As far as I remember resolver, refget proxy and metadata API are using the same error_response file that generates the same format I thought you're aware of it!  Let's go ahead with using the general format for now then since others are already using it.

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:73]: Done

File: app/vep/vep_resources.py
Comment (Mehrnaz-Charkhchi) [reply app/vep/vep_resources.py:81]: spec is updated.

**Review by azangru — APPROVED** [review]

File: app/vep/vep_resources.py
Comment (veidenberg) [inline app/vep/vep_resources.py:90]: `status.HTTP_500_INTERNAL_SERVER_ERROR` to make it consistent withg the other responses.

File: app/vep/vep_resources.py
Comment (veidenberg) [inline app/vep/vep_resources.py:66]: Are we using integer status codes or constants (`status.HTTP_500_INTERNAL_SERVER_ERROR`)?

File: app/vep/vep_resources.py
Comment (veidenberg) [reply app/vep/vep_resources.py:81]: I was confused why we're mixing integer and constant status codes in this file, but it seems that it's because we are reimplementing `http_404_not_found` function from  `response_error_handler` just to change the value of the `details` field. Would it make sense to instead update `response_error_handler` to accept a custom status message?

**Review by jyothishnt — APPROVED** [review]

**Pull request merged**
